nautobot/ipam/tests/test_api.py

```diff
@@ -1006,20 +1015,109 @@ def test_list_available_vlans(self):
         """
         Test retrieval of all available VLAN IDs within a VLANGroup.
         """
-        vg = VLANGroup.objects.create(name="Test", range="5-10,15-20")
-        VLAN.objects.create(name="vlan_5", vid=5, status=Status.objects.first(), vlan_group=vg)
-        VLAN.objects.create(name="vlan_10", vid=10, status=Status.objects.first(), vlan_group=vg)
-        VLAN.objects.create(name="vlan_17", vid=17, status=Status.objects.first(), vlan_group=vg)
-        unused_vids = [6, 7, 8, 9, 15, 16, 18, 19, 20]
-
-        url = reverse("ipam-api:vlangroup-available-vlans", kwargs={"pk": vg.pk})
+        url = reverse("ipam-api:vlangroup-available-vlans", kwargs={"pk": self.vlan_group.pk})
         self.add_permissions("ipam.view_vlangroup")
 
         # Retrieve all available VLAN IDs
         response = self.client.get(url, **self.header)
 
-        self.assertEqual(response.data["results"], unused_vids)
-        self.assertEqual(response.data["count"], len(unused_vids))
+        self.assertEqual(response.data["results"], self.unused_vids)
+        self.assertEqual(response.data["count"], len(self.unused_vids))
+
+    def test_create_single_available_vlan(self):
+        """
+        Test creation of the first available VLAN within a VLANGroup.
+        """
+        cf = CustomField.objects.create(key="sor", label="Source of Record Field", type="text")
+        cf.content_types.add(ContentType.objects.get_for_model(VLAN))
+        url = reverse("ipam-api:vlangroup-available-vlans", kwargs={"pk": self.vlan_group.pk})
+        self.add_permissions(
+            "ipam.view_vlangroup",
+            "ipam.change_vlangroup",
+            "ipam.add_vlangroup",
+            "ipam.change_vlan",
+            "ipam.view_vlan",
+            "ipam.add_vlan",
```

Comment:
I'm surprised all of these are necessary - I would have thought maybe just the `view` permissions plus `add_vlan` would be sufficient?

Reply:
Good point - Actually , the only one needed is `add_vlangroup` (tested).

Reply:
That sounds like a bug in waiting to me - it should be `add_vlan` even though this is a `vlangroup` endpoint.

Reply:
Fixed:
get - view_vlangroup
post - view_vlangroup, add_vlan 
permissions are required